File: solutions/java/rna-transcription/2/src/main/java/RnaTranscription.java
import java.util.ArrayList;
import java.util.List;

public class RnaTranscription {
    private final List<String> transcription = new ArrayList<>();

    String transcript(String character) {
        return character.equals("A") ? "U"
                : character.equals("T") ? "A"
                : character.equals("C") ? "G"
                : character.equals("G") ? "C"
                : "";
    }

    String transcribe(String dnaStrand) {
        for (char nucleotide: dnaStrand.toCharArray()) {
            transcription.add(transcript(String.valueOf(nucleotide)));
        }
        return String.join("", transcription);
    }
}

File: solutions/python/rna-transcription/2/rna_transcription.py
def translate(nucleotide):
    if nucleotide == "":
        return nucleotide
    if nucleotide == "C" : 
        return "G"
    if nucleotide == "G" : 
        return "C"
    if nucleotide == "A" : 
        return "U"
    if nucleotide == "T" : 
        return "A"
    return TypeError("Invalid nucleotide")

def to_rna(dna_strand):
    return ''.join([ translate(nucleotide) for nucleotide in dna_strand ])

